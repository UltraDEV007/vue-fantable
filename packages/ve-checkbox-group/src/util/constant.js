// prefix
export const PREFIX_CLS = 've-checkbox-group-'

// emit events
export const EMIT_EVENTS = {
  // on checked change
  ON_CHECKED_CHANGE: 'on-checked-change',
}

// comps name
export const COMPS_NAME = {
  VE_CHECKBOX: 'VeCheckbox',
  VE_CHECKBOX_GROUP: 'VeCheckboxGroup',
}
