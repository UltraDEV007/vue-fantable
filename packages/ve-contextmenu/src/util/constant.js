// prefix
export const PREFIX_CLS = 've-contextmenu-'

// comps name
export const COMPS_NAME = {
  VE_CONTEXTMENU: 'VeContextmenu',
}

// emit events
export const EMIT_EVENTS = {
  ON_NODE_CLICK: 'on-node-click',
}

// init data
export const INIT_DATA = {
  // PARENT_ID: 0,
  PARENT_DEEP: -1,
}

// contextmenu types
export const CONTEXTMENU_NODE_TYPES = {
  // Separator
  SEPARATOR: 'SEPARATOR',
}

// instance methods
export const INSTANCE_METHODS = {
  HIDE_CONTEXTMENU: 'hideContextmenu',
}
