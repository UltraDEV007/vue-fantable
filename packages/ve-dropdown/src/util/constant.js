// prefix
export const PREFIX_CLS = 've-dropdown-'

// emit events
export const EMIT_EVENTS = {
  FILTER_RESET: 'on-filter-reset',
  FILTER_CONFIRM: 'on-filter-confirm',
  VISIBLE_CHANGE: 'on-dropdown-visible-change',
  ITEM_SELECT_CHANGE: 'on-item-select-change',
}

// comps name
export const COMPS_NAME = {
  VE_DROPDOWN: 'VeDropdown',
}
