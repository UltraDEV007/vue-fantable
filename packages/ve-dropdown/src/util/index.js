/*
 * @clsName
 * @desc  get class name
 * @param {string} cls - class
 */
export function clsName(cls) {
  return 've-dropdown-' + cls
}
