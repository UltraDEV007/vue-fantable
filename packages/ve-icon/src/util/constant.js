// prefix
export const PREFIX_CLS = 've-icon-'

// emit events
export const EMIT_EVENTS = {}

// comps name
export const COMPS_NAME = {
  VE_ICON: 'VeIcon',
}
