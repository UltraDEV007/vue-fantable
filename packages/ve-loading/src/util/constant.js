// prefix
export const PREFIX_CLS = 've-loading-'

// emit events
export const EMIT_EVENTS = {}

// comps name
export const COMPS_NAME = {
  VE_LOADING: 'VeLoading',
  VE_LOADING_PLANE: 'VeLoadingPlane',
  /* VE_LOADING_CHASE: "VeLoadingChase", */
  VE_LOADING_BOUNCE: 'VeLoadingBounce',
  VE_LOADING_WAVE: 'VeLoadingWave',
  VE_LOADING_PULSE: 'VeLoadingPulse',
  VE_LOADING_FLOW: 'VeLoadingFlow',
  /* VE_LOADING_SWING: "VeLoadingSwing", */
  /*  VE_LOADING_CIRCLE: "VeLoadingCircle",
VE_LOADING_CIRCLE_FADE: "VeLoadingCircleFade", */
  VE_LOADING_GRID: 'VeLoadingGrid',
  VE_LOADING_FOLD: 'VeLoadingFold',
}

// spin name
export const SPIN_NAMES = {
  PLANE: 'plane',
  GRID: 'grid',
  WAVE: 'wave',
  FLOW: 'flow',
  BOUNCE: 'bounce',
  PULSE: 'pulse',
}
