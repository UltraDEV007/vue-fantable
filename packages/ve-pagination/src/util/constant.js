// prefix
export const PREFIX_CLS = 've-pagination-'

// locale comp name
export const LOCALE_COMP_NAME = 'pagination'

// emit events
export const EMIT_EVENTS = {
  PAGE_NUMBER_CHANGE: 'on-page-number-change',
  PAGE_SIZE_CHANGE: 'on-page-size-change',
}

// comps name
export const COMPS_NAME = {
  VE_PAGINATION: 'VePagination',
}
