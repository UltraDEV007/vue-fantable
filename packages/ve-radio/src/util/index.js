import { PREFIX_CLS } from './constant'

/*
 * @clsName
 * @desc  get class name
 * @param {string} cls - class
 */
export function clsName(cls) {
  return PREFIX_CLS + cls
}
