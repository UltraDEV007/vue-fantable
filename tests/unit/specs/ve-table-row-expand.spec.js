import { mount } from "@vue/test-utils";
import veTable from "@/ve-table";
import { later } from "../util";

describe("veTable row expand", () => {
    const TABLE_DATA = [
        {
            rowKey: 1001,
            name: "John",
            date: "1900-05-20",
            hobby: "coding",
            address: "No.1 Century Avenue, Shanghai",
        },
        {
            rowKey: 1002,
            name: "Dickerson",
            date: "1910-06-20",
            hobby: "coding",
            address: "No.1 Century Avenue, Beijing",
        },
        {
            rowKey: 1003,
            name: "Larsen",
            date: "2000-07-20",
            hobby: "coding and coding repeat",
            address: "No.1 Century Avenue, Chongqing",
        },
        {
            rowKey: 1004,
            name: "Geneva",
            date: "2010-08-20",
            hobby: "coding and coding repeat",
            address: "No.1 Century Avenue, Xiamen",
        },
        {
            rowKey: 1005,
            name: "Jami",
            date: "2020-09-20",
            hobby: "coding and coding repeat",
            address: "No.1 Century Avenue, Shenzhen",
        },
    ];

    const COLUMNS = [
        {
            field: "",
            key: "a",
            // 设置需要显示展开图标的列
            type: "expand",
            title: "",
            width: 50,
            align: "center",
        },
        {
            field: "name",
            key: "b",
            title: "Name",
            width: 200,
            align: "left",
        },
        {
            field: "hobby",
            key: "c",
            title: "Hobby",
            width: 300,
            align: "left",
        },
        {
            field: "address",
            key: "d",
            title: "Address",
            width: "",
            align: "left",
        },
    ];

    it("render", () => {
        const wrapper = mount(veTable, {
            propsData: {
                columns: COLUMNS,
                tableData: TABLE_DATA,
                expandOption: {
                    defaultExpandAllRows: false,
                    // 默认需要展开的行
                    defaultExpandedRowKeys: [1001, 1003],
                    expandable: ({ row, column, rowIndex }) => {
                        if (row["rowKey"] === 1002) {
                            return false;
                        }
                    },
                },
                rowKeyFieldName: "rowKey",
            },
        });

        expect(wrapper.html()).toMatchSnapshot();
    });

    it("trigger type default(icon)", async () => {
        const wrapper = mount(veTable, {
            propsData: {
                columns: COLUMNS,
                tableData: TABLE_DATA,
                expandOption: {
                    render: ({ row, column, rowIndex }, h) => {
                        return (
                            <p>
                                My name is
                                <span style="color:#1890ff;">{row.name}</span>
                                ,I'm living in {row.address}
                            </p>
                        );
                    },
                },
                rowKeyFieldName: "rowKey",
            },
        });

        // 默认不渲染节点
        expect(wrapper.find(".ve-table-expand-tr").exists()).toBe(false);

        wrapper
            .findAll(".ve-table-body-tr")
            .at(0)
            .find(".ve-table-row-expand-icon")
            .trigger("click");

        await later();

        expect(
            wrapper
                .findAll(".ve-table-expand-tr")
                .at(0)
                .find(".ve-table-expand-td .ve-table-expand-td-content")
                .exists(),
        ).toBe(true);

        wrapper
            .findAll(".ve-table-body-tr")
            .at(0)
            .find(".ve-table-row-expand-icon")
            .trigger("click");

        await later();

        // 折叠后销毁节点
        expect(wrapper.find(".ve-table-expand-tr").exists()).toBe(false);
    });

    it("trigger type:cell", async () => {
        const wrapper = mount(veTable, {
            propsData: {
                columns: COLUMNS,
                tableData: TABLE_DATA,
                expandOption: {
                    trigger: "cell",
                    render: ({ row, column, rowIndex }, h) => {
                        return (
                            <p>
                                My name is
                                <span style="color:#1890ff;">{row.name}</span>
                                ,I'm living in {row.address}
                            </p>
                        );
                    },
                },
                rowKeyFieldName: "rowKey",
            },
        });

        wrapper
            .findAll(".ve-table-body-tr")
            .at(0)
            .findAll(".ve-table-body-td")
            .at(0)
            .trigger("click");

        await later();

        expect(
            wrapper
                .findAll(".ve-table-expand-tr")
                .at(0)
                .find(".ve-table-expand-td .ve-table-expand-td-content")
                .exists(),
        ).toBe(true);
    });

    it("trigger type:row", async () => {
        const wrapper = mount(veTable, {
            propsData: {
                columns: COLUMNS,
                tableData: TABLE_DATA,
                expandOption: {
                    trigger: "row",
                    render: ({ row, column, rowIndex }, h) => {
                        return (
                            <p>
                                My name is
                                <span style="color:#1890ff;">{row.name}</span>
                                ,I'm living in {row.address}
                            </p>
                        );
                    },
                },
                rowKeyFieldName: "rowKey",
            },
        });

        wrapper.findAll(".ve-table-body-tr").at(0).trigger("click");

        await later();

        expect(
            wrapper
                .findAll(".ve-table-expand-tr")
                .at(0)
                .find(".ve-table-expand-td .ve-table-expand-td-content")
                .exists(),
        ).toBe(true);
    });

    it("expandOption expandable", async () => {
        const wrapper = mount(veTable, {
            propsData: {
                columns: COLUMNS,
                tableData: TABLE_DATA,
                expandOption: {
                    expandable: ({ row, column, rowIndex }) => {
                        if (row["rowKey"] === 1002) {
                            return false;
                        }
                    },
                    render: ({ row, column, rowIndex }, h) => {
                        return (
                            <p>
                                My name is
                                <span style="color:#1890ff;">{row.name}</span>
                                ,I'm living in {row.address}
                            </p>
                        );
                    },
                },
                rowKeyFieldName: "rowKey",
            },
        });

        expect(
            wrapper
                .findAll(".ve-table-body-tr")
                .at(1)
                .find(".ve-table-row-expand-icon")
                .exists(),
        ).toBe(false);
    });

    it("expandOption defaultExpandedRowKeys", async () => {
        const wrapper = mount(veTable, {
            propsData: {
                columns: COLUMNS,
                tableData: TABLE_DATA,
                expandOption: {
                    defaultExpandedRowKeys: [1001, 1003],
                    render: ({ row, column, rowIndex }, h) => {
                        return (
                            <p>
                                My name is
                                <span style="color:#1890ff;">{row.name}</span>
                                ,I'm living in {row.address}
                            </p>
                        );
                    },
                },
                rowKeyFieldName: "rowKey",
            },
        });

        expect(
            wrapper
                .findAll(".ve-table-body-tr")
                .at(0)
                .find(".ve-table-row-expand-icon")
                .exists(),
        ).toBe(true);

        expect(
            wrapper
                .findAll(".ve-table-body-tr")
                .at(2)
                .find(".ve-table-row-expand-icon")
                .exists(),
        ).toBe(true);
    });

    it("expandOption defaultExpandAllRows", async () => {
        const wrapper = mount(veTable, {
            propsData: {
                columns: COLUMNS,
                tableData: TABLE_DATA,
                expandOption: {
                    defaultExpandAllRows: true,
                    render: ({ row, column, rowIndex }, h) => {
                        return (
                            <p>
                                My name is
                                <span style="color:#1890ff;">{row.name}</span>
                                ,I'm living in {row.address}
                            </p>
                        );
                    },
                },
                rowKeyFieldName: "rowKey",
            },
        });

        expect(wrapper.findAll(".ve-table-row-expand-icon").length).toBe(5);
    });

    it("expandOption beforeExpandRowChange event", async () => {
        const mockFn = jest.fn();

        const wrapper = mount(veTable, {
            propsData: {
                columns: COLUMNS,
                tableData: TABLE_DATA,
                expandOption: {
                    render: ({ row, column, rowIndex }, h) => {
                        return (
                            <p>
                                My name is
                                <span style="color:#1890ff;">{row.name}</span>
                                ,I'm living in {row.address}
                            </p>
                        );
                    },
                    beforeExpandRowChange: ({
                        beforeExpandedRowKeys,
                        row,
                        rowIndex,
                    }) => {
                        if (row["rowKey"] === 1001) {
                            mockFn(false);
                            return false;
                        }
                        mockFn(true);
                        return true;
                    },
                },

                rowKeyFieldName: "rowKey",
            },
        });

        wrapper
            .findAll(".ve-table-body-tr")
            .at(0)
            .find(".ve-table-row-expand-icon")
            .trigger("click");
        expect(mockFn).toHaveBeenCalled();
        expect(mockFn).toHaveBeenCalledWith(false);

        await later();
        expect(wrapper.findAll(".ve-table-expand-tr").exists()).toBe(false);

        wrapper
            .findAll(".ve-table-body-tr")
            .at(1)
            .find(".ve-table-row-expand-icon")
            .trigger("click");
        expect(mockFn).toHaveBeenCalled();
        expect(mockFn).toHaveBeenCalledWith(true);

        await later();
        expect(
            wrapper
                .findAll(".ve-table-expand-tr")
                // 只要一行
                .at(0)
                .find(".ve-table-expand-td .ve-table-expand-td-content")
                .exists(),
        ).toBe(true);
    });

    it("expandOption afterExpandRowChange event", async () => {
        const mockFn = jest.fn();

        const wrapper = mount(veTable, {
            propsData: {
                columns: COLUMNS,
                tableData: TABLE_DATA,
                expandOption: {
                    render: ({ row, column, rowIndex }, h) => {
                        return (
                            <p>
                                My name is
                                <span style="color:#1890ff;">{row.name}</span>
                                ,I'm living in {row.address}
                            </p>
                        );
                    },
                    afterExpandRowChange: ({
                        afterExpandedRowKeys,
                        row,
                        rowIndex,
                    }) => {
                        mockFn();
                    },
                },

                rowKeyFieldName: "rowKey",
            },
        });

        wrapper
            .findAll(".ve-table-body-tr")
            .at(0)
            .find(".ve-table-row-expand-icon")
            .trigger("click");
        expect(mockFn).toHaveBeenCalled();
    });

    it("expandOption expandedRowKeys(controllable attr)", async () => {
        const mockFn = jest.fn();

        const wrapper = mount({
            template: `
            <div>
                <button class="button-demo" @click="expandRow(1003)">第3行展开切换</button>
                <ve-table
                style="width:100%"
                :columns="columns"
                :table-data="tableData"
                :expand-option="expandOption"
                row-key-field-name="rowKey"
                />
            </div>
            `,
            data() {
                return {
                    expandOption: {
                        expandedRowKeys: [1001], //this.expandRowKeys,
                        render: ({ row, column, rowIndex }, h) => {
                            return (
                                <p>
                                    My name is
                                    <span style="color:#1890ff;">
                                        {row.name}
                                    </span>
                                    ,I'm living in {row.address}
                                </p>
                            );
                        },
                        // 重新赋值处理
                        afterExpandRowChange: ({
                            afterExpandedRowKeys,
                            row,
                            rowIndex,
                        }) => {
                            mockFn(afterExpandedRowKeys);
                            this.changeExpandedRowKeys(afterExpandedRowKeys);
                        },
                    },
                    columns: COLUMNS,
                    tableData: TABLE_DATA,
                };
            },
            methods: {
                // 给可控属性重新赋值
                changeExpandedRowKeys(keys) {
                    this.expandOption.expandedRowKeys = keys;
                    mockFn(keys);
                },
                // 切换展开行
                expandRow(key) {
                    this.expandOption.expandedRowKeys.push(key);
                },
            },
        });

        expect(
            wrapper
                .findAll(".ve-table-expand-tr")
                .at(0)
                .find(".ve-table-expand-td .ve-table-expand-td-content")
                .exists(),
        ).toBe(true);

        wrapper.find(".button-demo").trigger("click");

        await later();

        expect(
            wrapper
                .findAll(".ve-table-expand-tr")
                .at(1)
                .find(".ve-table-expand-td .ve-table-expand-td-content")
                .exists(),
        ).toBe(true);

        wrapper
            .findAll(".ve-table-body-tr")
            .at(1)
            .find(".ve-table-row-expand-icon")
            .trigger("click");

        await later();

        expect(mockFn).toHaveBeenCalled();
        expect(mockFn).toHaveBeenCalledWith([1001, 1003, 1002]);
        expect(wrapper.html()).toMatchSnapshot();
    });
});
