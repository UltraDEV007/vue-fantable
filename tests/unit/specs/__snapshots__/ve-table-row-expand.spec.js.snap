// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`veTable row expand expandOption expandedRowKeys(controllable attr) 1`] = `
<div><button class="button-demo">第3行展开切换</button>
  <div class="vue-table-root" style="width: 100%;">
    <div class="fan-table fan-table-border-around" style="width: 100%;">
      <div class="fan-table-container fan-table-enable-cell-selection" style="height: 0px;">
        <div class="fan-table-content-wrapper">
          <table class="fan-table-content fan-table-border-x">
            <colgroup>
              <col style="width: 50px;">
              <col style="width: 200px;">
              <col style="width: 300px;">
              <col>
            </colgroup>
            <thead class="fan-table-fixed-header fan-table-header">
              <tr class="fan-table-header-tr">
                <th rowspan="1" colspan="1" class="fan-table-header-th" style="text-align: center; top: 0px;"></th>
                <th rowspan="1" colspan="1" class="fan-table-header-th" style="text-align: left; top: 0px;">Name</th>
                <th rowspan="1" colspan="1" class="fan-table-header-th" style="text-align: left; top: 0px;">Hobby</th>
                <th rowspan="1" colspan="1" class="fan-table-header-th fan-table-last-column" style="text-align: left; top: 0px;">Address</th>
              </tr>
            </thead>
            <tbody class="fan-table-body fan-table-row-hover fan-table-row-highlight">
              <tr style="height: 0px;">
                <td style="padding: 0px; border: 0px; height: 0px;"></td>
                <td style="padding: 0px; border: 0px; height: 0px;"></td>
                <td style="padding: 0px; border: 0px; height: 0px;"></td>
                <td style="padding: 0px; border: 0px; height: 0px;"></td>
              </tr>
              <tr row-key="1001" class="fan-table-body-tr">
                <td rowspan="1" colspan="1" col-key="a" class="fan-table-body-td" style="text-align: center;"><span class="fan-table-row-expand-icon fan-table-expand-icon-collapsed"><i class="ve-icon iconfont-vet icon-vet-right-arrow"></i></span></td>
                <td rowspan="1" colspan="1" col-key="b" class="fan-table-body-td" style="text-align: left;">
                  <!---->John
                </td>
                <td rowspan="1" colspan="1" col-key="c" class="fan-table-body-td" style="text-align: left;">
                  <!---->coding
                </td>
                <td rowspan="1" colspan="1" col-key="d" class="fan-table-body-td" style="text-align: left;">
                  <!---->No.1 Century Avenue, Shanghai
                </td>
              </tr>
              <tr class="fan-table-expand-tr">
                <td colspan="4" class="fan-table-expand-td">
                  <div class="fan-table-expand-td-content">
                    <p>My name is<span style="color: rgb(24, 144, 255);">John</span>,I'm living in No.1 Century Avenue, Shanghai</p>
                  </div>
                </td>
              </tr>
              <tr row-key="1002" class="fan-table-body-tr">
                <td rowspan="1" colspan="1" col-key="a" class="fan-table-body-td" style="text-align: center;"><span class="fan-table-row-expand-icon fan-table-expand-icon-collapsed"><i class="ve-icon iconfont-vet icon-vet-right-arrow"></i></span></td>
                <td rowspan="1" colspan="1" col-key="b" class="fan-table-body-td" style="text-align: left;">
                  <!---->Dickerson
                </td>
                <td rowspan="1" colspan="1" col-key="c" class="fan-table-body-td" style="text-align: left;">
                  <!---->coding
                </td>
                <td rowspan="1" colspan="1" col-key="d" class="fan-table-body-td" style="text-align: left;">
                  <!---->No.1 Century Avenue, Beijing
                </td>
              </tr>
              <tr class="fan-table-expand-tr">
                <td colspan="4" class="fan-table-expand-td">
                  <div class="fan-table-expand-td-content">
                    <p>My name is<span style="color: rgb(24, 144, 255);">Dickerson</span>,I'm living in No.1 Century Avenue, Beijing</p>
                  </div>
                </td>
              </tr>
              <tr row-key="1003" class="fan-table-body-tr">
                <td rowspan="1" colspan="1" col-key="a" class="fan-table-body-td" style="text-align: center;"><span class="fan-table-row-expand-icon fan-table-expand-icon-collapsed"><i class="ve-icon iconfont-vet icon-vet-right-arrow"></i></span></td>
                <td rowspan="1" colspan="1" col-key="b" class="fan-table-body-td" style="text-align: left;">
                  <!---->Larsen
                </td>
                <td rowspan="1" colspan="1" col-key="c" class="fan-table-body-td" style="text-align: left;">
                  <!---->coding and coding repeat
                </td>
                <td rowspan="1" colspan="1" col-key="d" class="fan-table-body-td" style="text-align: left;">
                  <!---->No.1 Century Avenue, Chongqing
                </td>
              </tr>
              <tr class="fan-table-expand-tr">
                <td colspan="4" class="fan-table-expand-td">
                  <div class="fan-table-expand-td-content">
                    <p>My name is<span style="color: rgb(24, 144, 255);">Larsen</span>,I'm living in No.1 Century Avenue, Chongqing</p>
                  </div>
                </td>
              </tr>
              <tr row-key="1004" class="fan-table-body-tr">
                <td rowspan="1" colspan="1" col-key="a" class="fan-table-body-td" style="text-align: center;"><span class="fan-table-row-expand-icon"><i class="ve-icon iconfont-vet icon-vet-right-arrow"></i></span></td>
                <td rowspan="1" colspan="1" col-key="b" class="fan-table-body-td" style="text-align: left;">
                  <!---->Geneva
                </td>
                <td rowspan="1" colspan="1" col-key="c" class="fan-table-body-td" style="text-align: left;">
                  <!---->coding and coding repeat
                </td>
                <td rowspan="1" colspan="1" col-key="d" class="fan-table-body-td" style="text-align: left;">
                  <!---->No.1 Century Avenue, Xiamen
                </td>
              </tr>
              <!---->
              <tr row-key="1005" class="fan-table-body-tr">
                <td rowspan="1" colspan="1" col-key="a" class="fan-table-body-td" style="text-align: center;"><span class="fan-table-row-expand-icon"><i class="ve-icon iconfont-vet icon-vet-right-arrow"></i></span></td>
                <td rowspan="1" colspan="1" col-key="b" class="fan-table-body-td" style="text-align: left;">
                  <!---->Jami
                </td>
                <td rowspan="1" colspan="1" col-key="c" class="fan-table-body-td" style="text-align: left;">
                  <!---->coding and coding repeat
                </td>
                <td rowspan="1" colspan="1" col-key="d" class="fan-table-body-td" style="text-align: left;">
                  <!---->No.1 Century Avenue, Shenzhen
                </td>
              </tr>
              <!---->
            </tbody>
            <tfoot class="fan-table-fixed-footer fan-table-footer"></tfoot>
          </table>
          <div class="fan-table-selection-wrapper">
            <div class="fan-table-selection-fixed-left"></div>
            <div class="fan-table-selection-middle"></div>
            <div class="fan-table-selection-fixed-right"></div>
          </div>
        </div>
      </div>
      <div class="fan-table-edit-input-container" style="top: 0px; left: 0px; height: 1px; z-index: -1; opacity: 0;"><textarea tabindex="-1" class="fan-table-edit-input"></textarea></div>
    </div>
  </div>
</div>
`;

exports[`veTable row expand render 1`] = `
<div class="vue-table-root">
  <div class="fan-table fan-table-border-around" style="width: 100%;">
    <div class="fan-table-container fan-table-enable-cell-selection" style="height: 0px;">
      <div class="fan-table-content-wrapper">
        <table class="fan-table-content fan-table-border-x">
          <colgroup>
            <col style="width: 50px;">
            <col style="width: 200px;">
            <col style="width: 300px;">
            <col>
          </colgroup>
          <thead class="fan-table-fixed-header fan-table-header">
            <tr class="fan-table-header-tr">
              <th rowspan="1" colspan="1" class="fan-table-header-th" style="text-align: center; top: 0px;"></th>
              <th rowspan="1" colspan="1" class="fan-table-header-th" style="text-align: left; top: 0px;">Name</th>
              <th rowspan="1" colspan="1" class="fan-table-header-th" style="text-align: left; top: 0px;">Hobby</th>
              <th rowspan="1" colspan="1" class="fan-table-header-th fan-table-last-column" style="text-align: left; top: 0px;">Address</th>
            </tr>
          </thead>
          <tbody class="fan-table-body fan-table-row-hover fan-table-row-highlight">
            <tr style="height: 0px;">
              <td style="padding: 0px; border: 0px; height: 0px;"></td>
              <td style="padding: 0px; border: 0px; height: 0px;"></td>
              <td style="padding: 0px; border: 0px; height: 0px;"></td>
              <td style="padding: 0px; border: 0px; height: 0px;"></td>
            </tr>
            <tr row-key="1001" class="fan-table-body-tr">
              <td rowspan="1" colspan="1" col-key="a" class="fan-table-body-td" style="text-align: center;"><span class="fan-table-row-expand-icon fan-table-expand-icon-collapsed"><i class="ve-icon iconfont-vet icon-vet-right-arrow"></i></span></td>
              <td rowspan="1" colspan="1" col-key="b" class="fan-table-body-td" style="text-align: left;">
                <!---->John
              </td>
              <td rowspan="1" colspan="1" col-key="c" class="fan-table-body-td" style="text-align: left;">
                <!---->coding
              </td>
              <td rowspan="1" colspan="1" col-key="d" class="fan-table-body-td" style="text-align: left;">
                <!---->No.1 Century Avenue, Shanghai
              </td>
            </tr>
            <tr class="fan-table-expand-tr">
              <td colspan="4" class="fan-table-expand-td">
                <div class="fan-table-expand-td-content"></div>
              </td>
            </tr>
            <tr row-key="1002" class="fan-table-body-tr">
              <td rowspan="1" colspan="1" col-key="a" class="fan-table-body-td" style="text-align: center;"></td>
              <td rowspan="1" colspan="1" col-key="b" class="fan-table-body-td" style="text-align: left;">Dickerson</td>
              <td rowspan="1" colspan="1" col-key="c" class="fan-table-body-td" style="text-align: left;">coding</td>
              <td rowspan="1" colspan="1" col-key="d" class="fan-table-body-td" style="text-align: left;">No.1 Century Avenue, Beijing</td>
            </tr>
            <tr row-key="1003" class="fan-table-body-tr">
              <td rowspan="1" colspan="1" col-key="a" class="fan-table-body-td" style="text-align: center;"><span class="fan-table-row-expand-icon fan-table-expand-icon-collapsed"><i class="ve-icon iconfont-vet icon-vet-right-arrow"></i></span></td>
              <td rowspan="1" colspan="1" col-key="b" class="fan-table-body-td" style="text-align: left;">
                <!---->Larsen
              </td>
              <td rowspan="1" colspan="1" col-key="c" class="fan-table-body-td" style="text-align: left;">
                <!---->coding and coding repeat
              </td>
              <td rowspan="1" colspan="1" col-key="d" class="fan-table-body-td" style="text-align: left;">
                <!---->No.1 Century Avenue, Chongqing
              </td>
            </tr>
            <tr class="fan-table-expand-tr">
              <td colspan="4" class="fan-table-expand-td">
                <div class="fan-table-expand-td-content"></div>
              </td>
            </tr>
            <tr row-key="1004" class="fan-table-body-tr">
              <td rowspan="1" colspan="1" col-key="a" class="fan-table-body-td" style="text-align: center;"><span class="fan-table-row-expand-icon"><i class="ve-icon iconfont-vet icon-vet-right-arrow"></i></span></td>
              <td rowspan="1" colspan="1" col-key="b" class="fan-table-body-td" style="text-align: left;">
                <!---->Geneva
              </td>
              <td rowspan="1" colspan="1" col-key="c" class="fan-table-body-td" style="text-align: left;">
                <!---->coding and coding repeat
              </td>
              <td rowspan="1" colspan="1" col-key="d" class="fan-table-body-td" style="text-align: left;">
                <!---->No.1 Century Avenue, Xiamen
              </td>
            </tr>
            <!---->
            <tr row-key="1005" class="fan-table-body-tr">
              <td rowspan="1" colspan="1" col-key="a" class="fan-table-body-td" style="text-align: center;"><span class="fan-table-row-expand-icon"><i class="ve-icon iconfont-vet icon-vet-right-arrow"></i></span></td>
              <td rowspan="1" colspan="1" col-key="b" class="fan-table-body-td" style="text-align: left;">
                <!---->Jami
              </td>
              <td rowspan="1" colspan="1" col-key="c" class="fan-table-body-td" style="text-align: left;">
                <!---->coding and coding repeat
              </td>
              <td rowspan="1" colspan="1" col-key="d" class="fan-table-body-td" style="text-align: left;">
                <!---->No.1 Century Avenue, Shenzhen
              </td>
            </tr>
            <!---->
          </tbody>
          <tfoot class="fan-table-fixed-footer fan-table-footer"></tfoot>
        </table>
        <div class="fan-table-selection-wrapper">
          <div class="fan-table-selection-fixed-left"></div>
          <div class="fan-table-selection-middle"></div>
          <div class="fan-table-selection-fixed-right"></div>
        </div>
      </div>
    </div>
    <div class="fan-table-edit-input-container" style="top: 0px; left: 0px; height: 1px; z-index: -1; opacity: 0;"><textarea tabindex="-1" class="fan-table-edit-input"></textarea></div>
  </div>
</div>
`;
